idea/src/org/jetbrains/kotlin/idea/quickfix/AddJvmNameAnnotationFix.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.quickfix
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import org.jetbrains.kotlin.diagnostics.Diagnostic
+import org.jetbrains.kotlin.idea.KotlinBundle
+import org.jetbrains.kotlin.idea.core.CollectingNameValidator
+import org.jetbrains.kotlin.idea.core.KotlinNameSuggester
+import org.jetbrains.kotlin.idea.util.addAnnotation
+import org.jetbrains.kotlin.idea.util.findAnnotation
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.*
+
+class AddJvmNameAnnotationFix(element: KtElement, private val jvmName: String) : KotlinQuickFixAction<KtElement>(element) {
```

Comment:
Shouldn't we replace `jvmName` with `functionName` and move the logic of `jvmName` inside `invoke`?

Reply:
If `KotlinNameSuggester.suggestNameByName` call is moved inside `invoke`, Tests fails here:
https://github.com/JetBrains/kotlin/blob/3a5f42cc5ea618dae04bcb954bb1fea738a54a4c/idea/tests/org/jetbrains/kotlin/idea/quickfix/AbstractQuickFixTest.kt#L170-L172

Should we add `AddJvmNameAnnotationFix` to `idea/testData/quickfix/allowResolveInWriteAction.txt`?

Reply:
You right, we shouldn't do this.